tests/test_properties.py

```diff
@@ -0,0 +1,51 @@
+from hypothesis import given, example
+from hypothesis import strategies as st
+from hypothesis.strategies import composite
+
+from bytestring_splitter import BytestringSplitter
+
+
+@given(st.binary(min_size=1))
+@example(b'hello world')
+def test_tiny_splitter_properties(bytestring):
+    splitter = BytestringSplitter(len(bytestring))
+    assert splitter(bytestring) == [bytestring]
+
+
+@composite
+def bytesting_and_trid_pattern(draw):
+
+    # Hypothesis
+    bytestring = draw(st.binary(min_size=3))
+    near_the_end = len(bytestring)-1
+    first_member_length = draw(st.integers(min_value=1, max_value=near_the_end))
+    remainder = len(bytestring) - first_member_length
+    second_member_length = draw(st.integers(min_value=1, max_value=remainder))
+
+    # Setup Pattern
+    header_pattern = (first_member_length, second_member_length)
+    pattern = (*header_pattern, len(bytestring)-sum(header_pattern))
+
+    return bytestring, pattern
+
+
+@given(bytesting_and_trid_pattern())
```

Comment:
Interesting that you call it here (and thus pass the tuple).  Not how I imagined such an interface working.

Reply:
I have not looked under the hood here, but `bytestring_and_triad_pattern` is decorated `@composite`, so perhaps we are getting something else?

From the docs:

> The composite decorator works by converting a function that returns one example into a function that returns a strategy that produces such examples